stdlib/jwt/src/main/ballerina/src/jwt/jwt_validator.bal

```diff
@@ -270,18 +272,21 @@ function validateJwtRecords(string jwt, JwtHeader jwtHeader, JwtPayload jwtPaylo
     }
     JwtSigningAlgorithm alg = <JwtSigningAlgorithm>jwtHeader?.alg;  // The `()` value is already validated.
 
-    JwtTrustStoreConfig|JwksConfig? signatureConfig = config?.signatureConfig;
-    if (signatureConfig is JwtTrustStoreConfig) {
-        _ = check validateSignatureByTrustStore(jwt, alg, signatureConfig);
-    } else if (signatureConfig is JwksConfig) {
+    JwksConfig? jwksConfig = config?.jwksConfig;
+    if (jwksConfig is JwksConfig) {
         string? kid = jwtHeader?.kid;
         if (kid is string) {
-            _ = check validateSignatureByJwks(jwt, kid, alg, signatureConfig);
+            _ = check validateSignatureByJwks(jwt, kid, alg, jwksConfig);
         } else {
             return prepareError("Key ID (kid) is not provided in JOSE header.");
         }
     }
 
+    JwtTrustStoreConfig? trustStoreConfig = config?.trustStoreConfig;
+    if (trustStoreConfig is JwtTrustStoreConfig) {
+        _ = check validateSignatureByTrustStore(jwt, alg, trustStoreConfig);
+    }
```

Comment:
I think we need to move this as else if statement

Reply:
Yes. Otherwise the signature can be validated twice. Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/24189/commits/dba0b304c31a8f674e20a08c8736638925d219c8

Reply:
Shall we check the kid first, code looks like, 
```ballerina
if (kid is string) {
  if (jwksConfig is JwksConfig) {
    // validate
  } else if (trustStoreConfig is JwtTrustStoreConfig) {
   // validate
  } else {
    // error 
 }
} else {
 if (trustStoreConfig is JwtTrustStoreConfig) {
   // validate
  } else {
    // error 
 }
}
```
current logic will some cases, I think

Reply:
Actually, there is a use case to pass the signature validation if, none of the `JwksConfig` or `JwtTrustStoreConfig` are provided. And there is another use case where kid is not given, but only `JwksConfig` is provided, so that we have to return an error message. Considering all, I updated the logic as
```ballerina
if (jwksConfig is JwksConfig) {
    if (kid is string) {
        // validate
    } else if (trustStoreConfig is JwtTrustStoreConfig) {
       // validate
    } else {
        // return error
    }
} else if (trustStoreConfig is JwtTrustStoreConfig) {
    // validate
}
// continue without signature validation
```
WDYT?

Reply:
Added following test cases which covers all the possibilities.
``` 
  * Test cases for JWT signature validation with TrustStore and JWK based authentication/authorization scenarios.
  *
  * The following 8 scenarios are tested here.
  * T (TRUE) - Parameter/Config is available.
  * F (FALSE) - Parameter/Config is not available.
  *
  * | Scenario |  kid  | JwksConfig | TrustStoreConfig | Test Status |
  * |----------|-------|------------|------------------|-------------|
  * |   1      |   T   |     T      |         T        |    PASS     |
  * |   2      |   T   |     T      |         F        |    PASS     |
  * |   3      |   T   |     F      |         T        |    PASS     |
  * |   4      |   T   |     F      |         F        |    PASS     |
  * |   5      |   F   |     T      |         T        |    PASS     |
  * |   6      |   F   |     T      |         F        |    FAIL     |
  * |   7      |   F   |     F      |         T        |    PASS     |
  * |   8      |   F   |     F      |         F        |    PASS     |
  *
  * Additionally, there are 4 more scenarios where kid is available, but it is invalid.
  * |   9      |   T   |     T      |         T        |    FAIL     |
  * |  10      |   T   |     T      |         F        |    FAIL     |
  * |  11      |   T   |     F      |         T        |    PASS     |
  * |  12      |   T   |     F      |         F        |    PASS     |
```